lib/zip_source_filep.c

```diff
@@ -423,10 +426,12 @@ read_file(void *state, void *data, zip_uint64_t len, zip_source_cmd_t cmd) {
 	    zip_error_set(&ctx->error, ZIP_ER_RENAME, errno);
 	    return -1;
 	}
-	mask = umask(022);
+#if defined(HAVE_UMASK) && defined(HAVE_CHMOD)
```

Comment:
Neither of these are currently defined. They need to be added to the top level `CMakeLists.txt` and `cmake-config.h.in`.
Also, the Windows case where `_chmod` and `_umask` exist needs to be handled -- there we add `define`s for chmod and umask (see `lib/compat.h`).